### PR TITLE
Replace termcolor with rich #623

### PR DESCRIPTION
**Pull request by Arker123**
Changes:-
- Replacing termcolor with rich
- Removed --outfile cli parameter
- Moved write function inline

Screenshots :-

1) Command --> `floss -v [executable name]`
<p float="left">

</p>

2) Command --> `floss -v [executable name] --color never]`
<p float="left">

</p>

3) Command --> `floss -v [executable name] > file`

<p float="left">

</p>

### CODE REVIEW

File: floss/main.py [hunk 1]
```diff
@@ -270,18 +271,16 @@ def set_color_config(color):
     colorama.just_fix_windows_console()
 
     if color == "always":
-        colorama.init(strip=False)
```
Comment (williballenthin) [inline floss/main.py:273]: lets remove the colorama import if its no longer used here. also the dependency in setup.py

File: floss/main.py [hunk 2]
```diff
@@ -466,7 +465,7 @@ def get_signatures(sigs_path):
     return paths
 
 
-def write(results: ResultDocument, json_: bool, verbose: Verbosity, quiet: bool, outfile: Optional[str]):
+def write(results: ResultDocument, json_: bool, verbose: Verbosity, quiet: bool, outfile: Optional[str], console):
```
Comment (williballenthin) [inline floss/main.py:468]: i wonder if we should remove the `--outfile` cli parameter and rely on people redirecting output like `floss > output.json`. there's getting to be a lot of arguments to this function.

File: floss/render/default.py [hunk 3]
```diff
@@ -158,67 +164,66 @@ def render_staticstrings(strings, ostream, verbose, disable_headers):
...
-            render_sub_heading(" FUNCTION at " + heading_style(f"0x{fva:x}"), len(data), ostream, disable_headers)
+            render_sub_heading(" FUNCTION at " + heading_style(f"0x{fva:x}"), len(data), console, disable_headers)
             rows = []
             for ds in data:
                 if ds.address_type == AddressType.STACK:
-                    offset_string = "[stack]"
+                    offset_string = "Stack"
                 elif ds.address_type == AddressType.HEAP:
-                    offset_string = "[heap]"
+                    offset_string = "Heap"
```
Comment (williballenthin) [inline floss/render/default.py:211]: can you explain this change?

File: floss/main.py [hunk 4]
```diff
@@ -466,17 +441,13 @@ def get_signatures(sigs_path):
     return paths
 
 
-def write(results: ResultDocument, json_: bool, verbose: Verbosity, quiet: bool, outfile: Optional[str]):
+def write(results: ResultDocument, json_: bool, verbose: Verbosity, quiet: bool, color):
```
Comment (wballenthin) [inline floss/main.py:444]: please provide a type hint for `color`

File: floss/render/default.py [hunk 5]
```diff
@@ -158,67 +155,73 @@ def render_staticstrings(strings, ostream, verbose, disable_headers):
...
             strings_by_functions[ds.decoding_routine].append(ds)
 
         for fva, data in strings_by_functions.items():
-            render_sub_heading(" FUNCTION at " + heading_style(f"0x{fva:x}"), len(data), ostream, disable_headers)
+            render_sub_heading(" FUNCTION at " + heading_style(f"0x{fva:x}"), len(data), console, disable_headers)
             rows = []
             for ds in data:
                 if ds.address_type == AddressType.STACK:
-                    offset_string = "[stack]"
+                    offset_string = "\[Stack]"
```
Comment (wballenthin) [inline floss/render/default.py:207]:
```suggestion
                    offset_string = "\[stack]"
```

File: floss/render/default.py [hunk 6]
```diff
@@ -158,67 +155,73 @@ def render_staticstrings(strings, ostream, verbose, disable_headers):
...
                 elif ds.address_type == AddressType.HEAP:
-                    offset_string = "[heap]"
+                    offset_string = "\[Heap]"
```
Comment (wballenthin) [inline floss/render/default.py:209]:
```suggestion
                    offset_string = "\[heap]"
```

File: setup.py [hunk 7]
```diff
@@ -68,7 +67,6 @@
             # type stubs for mypy
             "types-PyYAML==6.0.10",
             "types-tabulate==0.9.0.1",
-            "types-colorama==0.4.15.8",
```
Comment (wballenthin) [inline setup.py:71]: are there type hints for `rich` that we can include? or not necessary?

**Review by wballenthin — CHANGES_REQUESTED** [review]
minor changes requested and then looks good to me!

File: floss/main.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (wballenthin) [reply floss/main.py:444]: i like that. the logic of this function is trivial and yet it accepts so many parameters. moving it inline seems like a good simplification.

File: floss/main.py (hunk 2 above)
Comment (wballenthin) [reply floss/main.py]: we should ensure that when `--color` is not explicitly provided and the output is redirect that color is automatically disabled. i think `rich` does this, but please confirm.

**Comment by mr-tz** [comment]
Also looks good to me overall! Thanks!
Can you share a screenshot of the new output here?

**Review by mr-tz — APPROVED** [review]
lgtm

**Pull request merged**